PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/FilterBubbles.java

```diff
@@ -1,38 +1,27 @@
 package nl.tudelft.pl2016gr2.core.algorithms;
 
-import nl.tudelft.pl2016gr2.core.algorithms.subgraph.CompareSubgraphs;
 import nl.tudelft.pl2016gr2.model.graph.nodes.Bubble;
 import nl.tudelft.pl2016gr2.model.graph.nodes.GraphNode;
-import nl.tudelft.pl2016gr2.model.graph.nodes.PhyloBubble;
 import nl.tudelft.pl2016gr2.model.graph.nodes.PhyloFilter;
-import nl.tudelft.pl2016gr2.model.phylogenetictree.IPhylogeneticTreeNode;
-import nl.tudelft.pl2016gr2.model.phylogenetictree.IPhylogeneticTreeRoot;
-import nl.tudelft.pl2016gr2.visitor.BubbleChildrenVisitor;
 
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Iterator;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
-import java.util.Queue;
-import java.util.Set;
 
 /**
  * Class used to filter bubbles based on the phylogenetic tree.
  *
  * @author Casper
  *
  */
-public class FilterBubbles implements PhyloFilter {
+public abstract class FilterBubbles implements PhyloFilter {
```

Comment:
Rename `PhyloFilter` to `BubbleFilter` or `Filter` and call this class `Abstract<...>`

Also update the documentation for the old `PhyloFilter` interface.

Add the relevant methods (you specify some generic methods in this abstract class) to the interface rather than the abstract implementation (unless these are inherently different from other `PhyloFilter`s, which I don't think is the case)